can-reflect.js

```diff
@@ -19,9 +19,12 @@ var reflect = {};
 		reflect[prop] = reflections[prop];
 		//!steal-remove-start
 		if(typeof reflections[prop] === "function") {
-			Object.defineProperty(reflections[prop],"name",{
-				value: "canReflect."+prop
-			});
+			var propDescriptor = Object.getOwnPropertyDescriptor(reflections[prop], 'name');
+			if (propDescriptor && propDescriptor.writable && propDescriptor.configurable) {
```

Comment:
You might not have a descriptor for this property already.  I think this predicate should be 
`!propDescriptor || propDescriptor.writable && propDescriptor.configurable`